From 5bd28ffcf5bb20bc4f3b9c359d99dd2751fe6c93 Mon Sep 17 00:00:00 2001
From: Volker Lendecke <vl@samba.org>
Date: Mon, 19 Sep 2016 14:29:21 -0700
Subject: [PATCH] gencache: Bail out of stabilize if we can not get the
 allrecord lock

Bug: https://bugzilla.samba.org/show_bug.cgi?id=12045

Signed-off-by: Volker Lendecke <vl@samba.org>
Reviewed-by: Jeremy Allison <jra@samba.org>

Autobuild-User(master): Jeremy Allison <jra@samba.org>
Autobuild-Date(master): Tue Sep 20 04:09:33 CEST 2016 on sn-devel-144

(cherry picked from commit b208499960eefef02d305a3bd59b03a7c2aafcac)
---
 source3/lib/gencache.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/source3/lib/gencache.c b/source3/lib/gencache.c
index 84d273e..97e0b40 100644
--- a/source3/lib/gencache.c
+++ b/source3/lib/gencache.c
@@ -664,7 +664,7 @@ bool gencache_stabilize(void)
 		return false;
 	}
 
-	res = tdb_lockall(cache_notrans->tdb);
+	res = tdb_lockall_nonblock(cache_notrans->tdb);
 	if (res != 0) {
 		tdb_transaction_cancel(cache->tdb);
 		DEBUG(10, ("Could not get allrecord lock on "
-- 
2.9.3

